### PR TITLE
add eslint-visitor-keys

### PR DESCRIPTION
**Pull request by mysticatea**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/eslint-visitor-keys/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Type definitions for eslint-visitor-keys 1.0
+// Project: https://github.com/eslint/eslint-visitor-keys#readme
+// Definitions by: Toru Nagashima <https://github.com/mysticatea>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
```
Comment (DanielRosenwasser) [inline types/eslint-visitor-keys/index.d.ts]: Please remove this comment.

File: types/eslint-visitor-keys/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ *~ loaded outside a module loader environment, declare that global here.
+ *~ Otherwise, delete this declaration.
+ */
+export type VisitorKeys = Readonly<{
+    [type: string]: ReadonlyArray<string> | undefined
```
Comment (DanielRosenwasser) [inline types/eslint-visitor-keys/index.d.ts]: Mark this index signature as `readonly` instead of using the `Readonly` mapped type.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by mysticatea** [comment]
Thank you for the review. I updated this PR.

File: types/eslint-visitor-keys/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+// Definitions by: Toru Nagashima <https://github.com/mysticatea>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+export interface VisitorKeys {
+    readonly [type: string]: ReadonlyArray<string> | undefined;
+}
+
+export const KEYS: VisitorKeys;
+export function getKeys(node: {}): ReadonlyArray<string>;
```
Comment (DanielRosenwasser) [inline types/eslint-visitor-keys/index.d.ts]: This probable should be an estree Node. Feel free to follow up on this PR.

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**